### DISCUSSION TITLE
How to get more than one related collection through url?

### DISCUSSION THREAD

**Discussion by sandyksit** [directus/directus#12032] [Archive]
Hi @rijkvanzanten and team, I have create two collections and make Many to one relationship, 
like collections article and music,
I want to access parent collection (article) with included related collection (music), Could you please help me how to access from url ?
Now i am able to access http://localhost:8055/items/article, but also want to include music collection object, I don't want hit two times like separate items call.

Thanks,
Sandeep

**Comment by azrikahar** [comment]
You can use the `fields` query parameter 👍 Link to the docs: https://docs.directus.io/reference/query/#fields

To be more specific, based on the instructions/description in the readme and fitting it to your example, the URL may look like `http://localhost:8055/items/article?fields=*,music.*` (do change `music` to the actual relational column of your article data model). Do be mindful of the wildcards (`*`) and optimize it based on your needs, as described in the docs as well!

**Reply by sandyksit** [reply]
Thanks @azrikahar ,

**Reply by reqwire** [reply]
How do I do this if the M2A relationship is in another collection?

Example collections:
Products
Reviews

Reviews has M2A with Products.

I want to sort the products by the amount of reviews.

**Reply by azrikahar** [reply]
I believe you have found the answer as you mentioned here: https://github.com/directus/directus/discussions/11068#discussioncomment-3639923, but any feedback on how we can better clarify any of the pain points you have experienced in the docs will be very much appreciated 🙂

**Reply by reqwire** [reply]
> I believe you have found the answer as you mentioned here: [#11068 (reply in thread)](https://github.com/directus/directus/discussions/11068#discussioncomment-3639923), but any feedback on how we can better clarify any of the pain points you have experienced in the docs will be very much appreciated 🙂

I discovered if you set up a M2M, or M2A and want the data easilly accessible from another collection, you have to go into Advanced Mode where you can create a new field on the related collection. :D

Would be nice to have this in the docs somewhere because its a very useful feature

**Reply by azrikahar** [reply]
> I discovered if you set up a M2M, or M2A and want the data easilly accessible from another collection, you have to go into Advanced Mode where you can create a new field on the related collection. :D
> 
> Would be nice to have this in the docs somewhere because its a very useful feature

Thanks for the feedback! This does seem to crop up a few times where the Corresponding Field config isn't too apparent to newer users:

![chrome_DRuUbLyY2B](https://user-images.githubusercontent.com/42867097/190946385-c1ee254e-bdf7-4b12-a2b9-6288b2d6be27.png)

Upon further inspection in the docs, it seems like it is mentioned in this tip here: https://docs.directus.io/configuration/data-model/fields/relationship.html, but this page isn't being linked to from anywhere (found this by searching "Corresponding Field" in the whole repo directly). I believe it's slated for directus/docs#83 👍 cc @erondpowell

**Reply by erondpowell** [reply]
Interesting. @azrikahar  is right, this is an intentionally unlinked page, as it is draft for the coming 2nd "data model" pr where I describe fields in detail. I didn't realize Algolia would index all pages, even those unlinked in config.js ... @rijkvanzanten You want me to "clean these up" asap?

**Reply by erondpowell** [reply]
> Would be nice to have this in the docs somewhere because its a very useful feature

@reqwire If you get time, you mind skimming through these two sections and seeing if they address your point?
They are a conceptual intro. In addition to this existing documentation linked below, we are going to have a second BIG data model PR to improve the docs on fields pretty soon!

**M2M:** https://docs.directus.io/configuration/data-model/relationships.html#many-to-many-m2m
**M2A:** https://docs.directus.io/configuration/data-model/relationships.html#many-to-any-m2a

**Reply by lessborders** [reply]
This is muuuuuch more informative than the current docs, but it still doesn't mention the ability to create an additional field on the foreign model ;)

**Reply by lessborders** [reply]
Which is an incredibly useful feature/piece of information that would save someone headaches wondering why they can't query fields: "relationship.*"

**Reply by azrikahar** [reply]
>  I didn't realize Algolia would index all pages, even those unlinked in config.js

@erondpowell I believe it doesn't! When I mentioned finding this by searching the term `Corresponding Field` in the whole repo directly, I meant opening `directus/docs` repo locally and doing a "find all" in VS Code, rather than using the Algolia search in the official `docs.directus.io` site. 👍
